doc/source/working_with_kiwi/shell_scripts.rst

```diff
@@ -0,0 +1,422 @@
+   :file:`config.sh` and :file:`image.sh`, which can be used to further
+   customize an image in ways that are not possible via the image
+   description alone.
+
+
+KIWI supports up to two user defined scripts that it runs in the change
+root environment (chroot) containing your new appliance:
+
+1. :file:`config.sh` runs the end of the :ref:`prepare step <prepare-step>`
+   if present. It can be used to fine tune the unpacked image.
+
+2. :file:`images.sh` is executed at the beginning of the :ref:`image
+   creation process <create-step>`. It is run on the top level of the
+   target root tree. The script is usually used to remove files that are
+   not needed in the final image. For example, if an appliance is being
+   built for a specific hardware, unnecessary kernel drivers can be removed
+   using this script.
+
+KIWI will execute both scripts via the operating system if their executable
+bit is set (in that case a shebang is mandatory) otherwise they will be
+invoked via the BASH.
+
+.. _image-customization-config-sh:
+
+Image Customization via the ``config.sh`` Shell Script
+------------------------------------------------------
+
+The KIWI image description allows to have an :file:`config.sh` script in
+place. It can be used for changes appropriate for all images to be created
+from a given unpacked image (:file:`config.sh` runs prior to the *create
+step*). The script should add operating system configuration files which
+would be otherwise added by a user driven installer, like the activation of
+services, creation of configuration files, preparation of an environment
+for a firstboot workflow, etc.
+
+The :file:`config.sh` script is called at the end of the :ref:`prepare step
+<prepare-step>` (after users have been set and the *overlay tree directory*
+has been applied). If :file:`config.sh` exits with a non-zero exit code
+then KIWI will report the failure and abort the image creation.
+
+Find a common template for `config.sh` script below:
+
+.. code:: bash
+
+   #======================================
+   # Include functions & variables
+   #--------------------------------------
+   test -f /.kconfig && . /.kconfig
+   test -f /.profile && . /.profile
+
+   #======================================
+   # Greeting...
+   #--------------------------------------
+   echo "Configure image: [$kiwi_iname]..."
+
+   #======================================
+   # Mount system filesystems
+   #--------------------------------------
+   baseMount
+
+   #======================================
+   # Call configuration code/functions
+   #--------------------------------------
+   ...
+
+   #======================================
+   # Umount kernel filesystems
+   #--------------------------------------
+   baseCleanMount
+
+   #======================================
+   # Exit successfully
+   #--------------------------------------
+   exit 0
+
+Configuration Tips
+^^^^^^^^^^^^^^^^^^
+
+#. **Stateless systemd UUIDs:**
```

Comment:
I wrote that part quite a while ago... I am wondering if it is still the case. @dcermak just in case, have it faced or tested it? I'll try to verify this is still valid after going through this PR.

Reply:
No I haven't faced it at all, I just took that from the existing docs. Not sure if I even rephrased this, so it might very well be your original text.